bgpd/bgp_zebra.c

```diff
@@ -356,6 +359,28 @@ static int bgp_interface_address_delete(ZAPI_CALLBACK_ARGS)
 		bgp_connected_delete(bgp, ifc);
 	}
 
+	addr = ifc->address;
```

Comment:
Should we move all this inside `bgp_connected_delete`?

Reply:
I went down that path at first but after looking at all the places it was called I was hesitant to make that big of a change.  The bug I was fixing was specific to this call path, hence it got included in this path.  I just wanted to limit the size of the destruction that could happen if this was wrong.